Assets/Scripts/Framework/CoalOvenSystem/CoalOven.cs

```diff
@@ -0,0 +1,120 @@
+using FishNet.Object;
+using System;
+using UnityEngine;
+
+using DerailedDeliveries.Framework.Train;
+using DerailedDeliveries.Framework.Utils;
+
+namespace DerailedDeliveries.Framework.CoalOvenSystem
+{
+    /// <summary>
+    /// A class responsible for controlling the coal oven.
+    /// </summary>
+    public class CoalOven : NetworkAbstractSingleton<CoalOven>
+    {
+        /// <summary>
+        /// Invoked when the coal amount changes.
+        /// </summary>
+        public Action<float> OnCoalAmountChanged;
```

Comment:
```suggestion
        public Action<float> onCoalAmountChanged;
```

Also move down to the other public declared vars

Reply:
The naming is correct according to the [convention](https://github.com/lucasvdiepen/DerailedDeliveries/wiki/Code-Conventions#events).